src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Cookies.cs

```diff
@@ -627,6 +627,7 @@ public static IEnumerable<object[]> CookieNamesValuesAndUseCookies()
                 yield return new object[] { "ABC", "123", useCookies };
                 yield return new object[] { "Hello", "World", useCookies };
                 yield return new object[] { "foo", "bar", useCookies };
+                yield return new object[] { "Hello World", "value", useCookies };
```

Comment:
We should add more test cases here for leading and trailing spaces in the cookie names.  What does the HTTP stack do when it receives headers such as 'Set-Cookie:       name    = value'

Reply:
I tried " name " and these tests break as the cookie names are trimmed somewhere:
GetAsync_SetCookieContainer_CookieSent
```cs
GetCookieHeaderValue(cookieName, cookieValue) = " name =value", requestData.GetSingleHeaderValue("Cookie") = "name =value"
```
GetAsync_ReceiveSetCookieHeader_CookieAdded
```cs
Assert.Equal(cookieName, collection[0].Name); // cookieName = " name ", collection[0].Name = "name"
```

Reply:
Should probably not allow leading or trailing spaces if its not a round-trippable name.

Reply:
I would agree, especially as Chrome at-least does trim. (See above)

I'll push another commit that does that unless I hear otherwise.